litho-it/src/test/java/com/facebook/litho/widget/TextSpecTest.java

```diff
@@ -345,4 +350,249 @@ private static Layout setupWidthTestTextLayout() {
+        null,
+        TextAlignment.LAYOUT_START))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.LAYOUT_START))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.LAYOUT_START))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
+  }
+
+  @Test
+  public void testTextAlignment_layoutEnd() {
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.LAYOUT_END))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.LAYOUT_END))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.LAYOUT_END))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.LAYOUT_END))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+  }
+
+  @Test
+  public void testTextAlignment_left() {
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.LEFT))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.LEFT))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.LEFT))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.LEFT))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+  }
+
+  @Test
+  public void testTextAlignment_right() {
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.RIGHT))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "asdf",
+        YogaDirection.RTL,
+        null,
+        TextAlignment.RIGHT))
+        .isEqualTo(Layout.Alignment.ALIGN_OPPOSITE);
+
+    assertThat(getMountedDrawableLayoutAlignment(
+        "من اليمين الى اليسار",
+        YogaDirection.LTR,
+        null,
+        TextAlignment.RIGHT))
+        .isEqualTo(Layout.Alignment.ALIGN_NORMAL);
```

Comment:
Javadoc for `LEFT` and `RIGHT` says that they don't respect text/locale prefs. Here in the test `RIGHT` doesn't respect `YogaDiraction`, but does respect Arabic's RTL nature. 
Am I wrong or this doesn't correspond to the javadoc?

Reply:
RIGHT means always right, but since the only public alignment enums we can use are NORMAL and OPPOSITE, we have to choose based on the directionality of the text itself. 

e.g. for RIGHT:
NORMAL for RTL text
OPPOSITE for LTR text

to make sure that the alignment is always RIGHT, regardless of text direction.

Reply:
I think Sergey's referring to the docs for TextAlignment that was added that say "Align the text to the left, ignoring the text or locale preferences for LTR/RTL.", but that doesn't seem to be reflected in these tests

Reply:
Nah I understood him perfectly, and I’m trying to explain that this test verified exactly what the docs intend.

English text + ALIGN_OPPOSITE == right aligned
Arabic text + ALIGN_NORMAL == right aligned

^ that’s what this test checks for

Reply:
Yoga direction is completely ignored, as it should be.

Reply:
Ah gotcha, it's like... double-factoring it in and therefore canceling it out :P

Reply:
Yeah it’s pretty confusing lol. Sucks that ALIGN_RIGHT and ALIGN_LEFT are hidden in the enum... otherwise we could just use those directly.

Reply:
Thanks for clarification, Vince. This double-factoring was confusing, was expecting that NORMAL and OPPOSITE don't change depending on the text locale.